ruby/ext/google/protobuf_c/storage.c

```diff
@@ -178,9 +178,42 @@ void native_slot_set_value_and_case(const char* name,
       if (CLASS_OF(value) == CLASS_OF(Qnil)) {
         value = Qnil;
       } else if (CLASS_OF(value) != type_class) {
-        rb_raise(cTypeError,
-                 "Invalid type %s to assign to submessage field '%s'.",
-                rb_class2name(CLASS_OF(value)), name);
+        // check for possible implicit conversions
+        VALUE converted_value = NULL;
+        char* field_type_name = rb_class2name(type_class);
+        char* value_type_name = rb_class2name(CLASS_OF(value));
+
+        // Time -> Google::Protobuf::Timestamp
+        if (strcmp(field_type_name, "Google::Protobuf::Timestamp") == 0 &&
+            strcmp(value_type_name, "Time") == 0) {
```

Comment:
Does ruby provide class instance of Time?

Reply:
Yes, I've updated to use the native check.